### DISCUSSION TITLE
Does TinTin++ internal #map function support to get the neighbors of the designated room?

### DISCUSSION THREAD

**Discussion by killunix** [scandum/tintin#64] [Script help]
Does TinTin++ internal #map function support to get the neighbors of the designated room?  like the following map, I would like to choose Room1 Room2 Room5 Room6 Room7 Room8 Room10 and Room11 as the search destinations, what is the best practice for this scenario by using the TT MAP? any help would be appreciated, thanks! 
-----------------------------------------------------

                                             Room12
                                                ｜  
                                             Room11
                                                ｜
                                             Room10            
                                                ｜     
                          Room4--Room5--Room6--Room--Room1--Room2-Room3   
                                                ｜     
                                              Room7
                                                ｜ 
                                              Room8
                                                ｜
                                              Room9

**Comment by scandum** [comment]
There's currently no option to specify a maximum search distance.

If you want to create a list of everything in a 3 room radius, the following should work.
```
#map list {variable}{test}
#list test indexate distance
#list test {filter} {{[1-7]}.%*}
```
The weighted distance of a room increases by 2 per room by default. By 1.0 for the room weight, and by another 1.0 for the exit weight.

**Comment by killunix** [comment]
Thanks for your support. But it looks like add all rooms in the roomarea to the list. could have some way to add one of the designated room's neighbors?

**Reply by scandum** [reply]
I added a note on my todo list to look into it, if possible I'll add such an option to a future release.

**Reply by killunix** [reply]
Good, thanks. I think to search in the designated small territory is more efficient than searching all city rooms.

**Reply by bugfly66** [reply]
> I added a note on my todo list to look into it, if possible I'll add such an option to a future release.

Is it supported now?

**Reply by scandum** [reply]
Kind of, here's an example for the latest release:

```
#ALIAS {test}
{
    #map list {} {variable} {test};
    #list test indexate distance;
    #list test refine {&0 > 0 && &0 < 10};
    #variable test
}
```
As mentioned earlier, the distance is the room weight, which increases 2 per room by default.

There is still no range limit you can add to the actual search, but tintin should automatically return the nearest room in a search.
